pyanaconda/modules/payload/live/live_os.py

```diff
@@ -99,44 +120,28 @@ def space_required(self):
         """
         return get_dir_size("/") * 1024
 
-    def detect_live_os_base_image(self):
-        """Detect live os image in the system."""
-        log.debug("Trying to detect live os base image automatically")
-        for block_device in ["/dev/mapper/live-base", "/dev/mapper/live-osimg-min"]:
-            try:
-                if stat.S_ISBLK(os.stat(block_device)[stat.ST_MODE]):
-                    log.debug("Detected live base image %s", block_device)
-                    return block_device
-            except FileNotFoundError:
-                pass
-
-        # Is it a squashfs+overlayfs base image?
-        if os.path.exists("/run/rootfsbase"):
-            try:
-                block_device = execWithCapture("findmnt", ["-n", "-o", "SOURCE", "/run/rootfsbase"]).strip()
-                if block_device:
-                    log.debug("Detected live base image %s", block_device)
-                    return block_device
-            except (OSError, FileNotFoundError):
-                pass
-
-        log.debug("No live base image detected")
-        return ""
-
-    def setup_installation_source_with_task(self):
-        """Setup installation source device."""
-        return SetupInstallationSourceTask(self.image_path, INSTALL_TREE)
-
-    def teardown_installation_source_with_task(self):
+    def setup_installation_source_with_tasks(self):
+        """Setup installation source."""
+        self._check_source_availability("Set up source failed - source is not set!")
```

Comment:
Readiness check is moved now but availability is still required because the sources are passed in the task. So you want to know if you have what to pass in.

Reply:
You will know it just a little later when you expect these types of errors. We never expect that the task creation will fail.